microsetta_public_api/models/_taxonomy.py

```diff
@@ -137,6 +138,14 @@ def __init__(self, table: biom.Table, features: pd.DataFrame,
             formatter: Formatter = GreengenesFormatter()
         self._formatter = formatter
 
+        # initialize taxonomy tree
+        tree_data = ((i, lineage.split('; '))
+                     for i, lineage in self._features['Taxon'].items())
+        self.taxonomy_tree = skbio.TreeNode.from_taxonomy(tree_data)
+        for node in self.taxonomy_tree:
```

Comment:
Should this be `.traverse()`? `__iter__` only iterates over the children of the current node

Reply:
Good catch